hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java

```diff
@@ -64,6 +67,28 @@ public void testAddStorage() throws Exception {
     Assert.assertEquals(storage, blockInfo.getStorageInfo(0));
   }
 
+  @Test
+  public void testAddProvidedStorage() throws Exception {
+    BlockInfo blockInfo = new BlockInfoContiguous((short) 3);
+
+    DatanodeStorageInfo storage = mock(DatanodeStorageInfo.class);
+    when(storage.getStorageType()).thenReturn(StorageType.PROVIDED);
+    boolean added = blockInfo.addStorage(storage, blockInfo);
+
+    Assert.assertTrue(added);
+    Assert.assertEquals(storage, blockInfo.getStorageInfo(0));
+    Assert.assertTrue(blockInfo.isProvided());
+
```

Comment:
What about splitting this test into two with a name that describes what we are testing in each?
Another option would be to just have a different blockInfo for the second chunk.

Reply:
There exists a test for verifying behavior of add storage, `testAddStorage `. I had originally modified the original test and added the `isProvided` check. It seemed a bit confusing and out of place to me. So I added this test to clearly test both the scenarios. I would prefer this over two split methods. 
I can rename the second blockInfo if that adds value.

Reply:
I'm more concerned about readability. Renaming the second one is fine with me.

Reply:
Thanks !
I renamed it and added one more check.

Reply:
The test now appears splittable. Give me a sec and I will improve it.

Reply:
Done. Please take a look.

Reply:
LGTM